### DISCUSSION TITLE
How to write an analyzer using vscode or Visual Studio for Mac?

### DISCUSSION THREAD

**Discussion by hannahchan** [dotnet/roslyn#63920] [Q&A]
I want to enhance a library I've written with analyzer warnings however I can't seem to figure out how to get started using vscode or Visual Studio for Mac. Every documentation I find on the docs.microsoft.com website seems require Visual Studio for Windows. I currently don't have a Windows computer to use and I'm doing this as a hobby.

Are there any workaround or resources I could use to create my enhancements to my library?

**Comment by taori** [comment]
Clone this and see if you can compile it with vscode and probably dotnet cli. https://github.com/DotNetAnalyzers/StyleCopAnalyzers/tree/master/StyleCop.Analyzers

As far as i am aware there should be no issue developing with other platforms - done it with rider too.

**Reply by hannahchan** [reply]
Hi @taori,

The problem I have is I don't know how to get started. All the tutorials I can find including the [offical one](https://learn.microsoft.com/visualstudio/extensibility/getting-started-with-roslyn-analyzers) basically state that you need to install the `.NET Complier Platform` via Visual Studio.

![image](https://user-images.githubusercontent.com/22674140/197518149-de123be5-18e2-42b5-a7de-9df6d0f3b57d.png)

Because I'm on a Mac, I'm unable to do this and can't seem to do this with Visual Studio for Mac as well. Ideally I should be able to run `dotnet new` to create my analyzer from a template.

Once the project has been created, I don't think there is a problem to develop it using different platforms or IDEs. I did clone the StyleCop repository and get errors in the build however I do believe these are resolvable. However this then make it really hard for me to figure out how to build one especially if you've never built an analyzer before.

**Reply by taori** [reply]
@sharwell the templates in Vs is probably these right? https://github.com/dotnet/roslyn-sdk/tree/main/src/VisualStudio.Roslyn.SDK/Roslyn.SDK/ProjectTemplates/CSharp/Diagnostic these are the templates

Is it a requirement that they are VSTemplates or are there any objections to making them dotnet new templates?

**Reply by jasonmalinowski** [reply]
@jmarolf Any reason we can't make these work in dotnet new too?

**Reply by jmarolf** [reply]
We have a PR out for this but haven't merged it yet: https://github.com/dotnet/roslyn-sdk/pull/612
